crates/clarirs_core/src/ast/op.rs

```diff
@@ -144,29 +144,72 @@ impl<'c> AstOp<'c> {
             | AstOp::SLE(lhs, rhs)
             | AstOp::SGT(lhs, rhs)
             | AstOp::SGE(lhs, rhs) => lhs.kind().is_bitvec() && rhs.kind().is_bitvec(),
-            AstOp::ZeroExt(_, _) => todo!(),
-            AstOp::SignExt(_, _) => todo!(),
-            AstOp::Extract(_, _, _) => todo!(),
-            AstOp::Reverse(_) => todo!(),
-            AstOp::FpToFp(_, _) => todo!(),
-            AstOp::BvToFpUnsigned(_, _, _) => todo!(),
-            AstOp::FpToIEEEBV(_) => todo!(),
-            AstOp::FpToUBV(_, _, _) => todo!(),
-            AstOp::FpToSBV(_, _, _) => todo!(),
+            AstOp::ZeroExt(input_bitvec, _num_of_zeroes) => input_bitvec.kind().is_bitvec(),
+            AstOp::SignExt(input_bitvec, _num_of_zeroes) => input_bitvec.kind().is_bitvec(),
+            AstOp::Extract(input_bitvec, _from, _to) => input_bitvec.kind().is_bitvec(),
+            AstOp::Reverse(input_bitvec) => input_bitvec.kind().is_bitvec(),
+            AstOp::FpToFp(floating_point, sort) => {
+                floating_point.kind().is_float() && matches!(sort, FSort { .. })
+            }
+            AstOp::BvToFpUnsigned(floating_point, sort, rounding_mode) => {
+                floating_point.kind().is_float()
+                    && matches!(sort, FSort { .. })
+                    && matches!(
+                        rounding_mode,
+                        FPRM::NearestTiesToEven
```

Comment:
If all FPRMs apply (which I think is always true?), then like before we can just trust the compiler that it is a valid FPRM